### PR TITLE
Person.java: update isSamePerson method to check for email and phone

### LINKED ISSUE #50: Adjust equality comparator for Person class
The `Person` class checks equality based only on the name, but it is possible for multiple people to have the same name. Therefore it is better to check for equality based on the name, email and phone number combined.

### PR DESCRIPTION
**Pull request by Dinoman44**
Fixes #50

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -92,7 +92,9 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         return otherPerson != null
-                && otherPerson.getName().equals(getName());
+                && otherPerson.getName().equals(getName())
```
Comment (JunLongling) [inline src/main/java/seedu/address/model/person/Person.java:95]:
```suggestion
                && otherPerson.getName().toString().equalsIgnoreCase(getName().toString())
```

Comment (JunLongling) [inline src/main/java/seedu/address/model/person/Person.java:95]:
```suggestion
                && otherPerson.getName().fullName.equalsIgnoreCase(getName().fullName)
```

**Review by JunLongling — COMMENTED** [review]
Since the names should be treated as case-insensitive, the current implementation could lead to duplicate entries (e.g., treating "John" and "john" as different people)

File: src/main/java/seedu/address/model/person/Person.java (hunk 1 above)
Comment (JunLongling) [inline src/main/java/seedu/address/model/person/Person.java:95]:
```suggestion
                && otherPerson.getName().toString().equalsIgnoreCase(getName().toString())
```

Comment (Dinoman44) [reply src/main/java/seedu/address/model/person/Person.java:95]: I'll modify the `Name` class for this.

**Review by JunLongling — APPROVED** [review]

**Pull request merged**